File: arkham-horror-backend/arkham-core/library/Arkham/Types/SkillType.hs
module Arkham.Types.SkillType where

import ClassyPrelude
import Data.Aeson

data SkillType
  = SkillWillpower
  | SkillIntellect
  | SkillCombat
  | SkillAgility
  | SkillWild
  deriving stock (Show, Eq, Generic)
  deriving anyclass (ToJSON, FromJSON)

File: arkham-horror-frontend/src/arkham/types/SkillType.ts
import { JsonDecoder } from 'ts.data.json';

export type SkillType = 'SkillWillpower' | 'SkillIntellect' | 'SkillCombat' | 'SkillAgility' | 'SkillWild'

export const skillTypeDecoder = JsonDecoder.oneOf<SkillType>([
  JsonDecoder.isExactly('SkillWillpower'),
  JsonDecoder.isExactly('SkillIntellect'),
  JsonDecoder.isExactly('SkillCombat'),
  JsonDecoder.isExactly('SkillAgility'),
  JsonDecoder.isExactly('SkillWild'),
], 'SkillType');
